### PR TITLE
[SYCL] Disable tests failing in CI

### PR DESCRIPTION
**Pull request by vladimirlaz**

### CODE REVIEW

File: SYCL/Matrix/joint_matrix_tensorcore_double.cpp [hunk 1]
```diff
@@ -1,4 +1,5 @@
-// REQUIRES: gpu, cuda
+// REQUIRES: gpu, cuda, TEMPORARY_DISABLED
+// The test is failing in CI and is under rework intel/llvm-test-suite#650.
```
Comment (bader) [inline SYCL/Matrix/joint_matrix_tensorcore_double.cpp]: This is covered already by https://github.com/intel/llvm-test-suite/pull/667.

File: SYCL/Basic/free_function_queries/free_function_queries.cpp [hunk 2]
```diff
@@ -4,6 +4,9 @@
 // RUN: %CPU_RUN_PLACEHOLDER %t.out
 // RUN: %GPU_RUN_PLACEHOLDER %t.out
 
+// REQUIRED: TEMPORARY_DISABLED
```
Comment (bader) [inline SYCL/Basic/free_function_queries/free_function_queries.cpp]: Why not UNSUPPORTED: windows?
I think you mentioned that it does pass on Linux. Right?

Comment (bader) [reply SYCL/Basic/free_function_queries/free_function_queries.cpp]: BTW, I think we should file a bug report for missing compiler builds on Windows.

Comment (vladimirlaz) [reply SYCL/Basic/free_function_queries/free_function_queries.cpp]: Yes, but it is not guaranteed for all compilers.

Comment (bader) [reply SYCL/Basic/free_function_queries/free_function_queries.cpp]: > Yes, but it is not guaranteed for all compilers.

Which compilers in addition to oneAPI DPC++ do you mean? 
I think from now on these tests should pass on Linux. Am I wrong?

Comment (bader) [reply SYCL/Basic/free_function_queries/free_function_queries.cpp]: Take a look at pre-commit results for https://github.com/intel/llvm-test-suite/pull/667. Linux is clean and all these tests are failing on Windows only.

**Review by bader — APPROVED** [review]
Thanks!
LGTM, although I would check with @alexbatashev if XFAIL is the right way to disable SYCL/XPTI/basic_event_collection.cpp or we should also apply UNSUPPORTED to be able to fix the compiler w/o breaking CI (if needed).

**Review by alexbatashev — DISMISSED** [review]

File: SYCL/XPTI/basic_event_collection.cpp [hunk 3]
```diff
@@ -3,6 +3,8 @@
 // RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %s -o %t.out
 // RUN: env XPTI_TRACE_ENABLE=1 env XPTI_FRAMEWORK_DISPATCHER=%xptifw_dispatcher env XPTI_SUBSCRIBERS=%t_collector.dll env SYCL_DEVICE_FILTER=opencl %t.out | FileCheck %s 2>&1
 
+// It looks like order of events diffres on Windows
+// XFAIL: windows
```
Comment (alexbatashev) [inline SYCL/XPTI/basic_event_collection.cpp]: I think writing a single test here gets too complex. We should probably simply have two distinct files for each of OSes, like we do for some other tests.

Comment (bader) [inline SYCL/XPTI/basic_event_collection.cpp]:
```suggestion
// REQUIRES: TEMPORARY_DISABLED
```
This test failed in pre-commit for this PR on Linux(!).

**Review by bader — CHANGES_REQUESTED** [review]

**Review by alexbatashev — DISMISSED** [review]

File: SYCL/XPTI/basic_event_collection.cpp (hunk 3 above)
Comment (vladimirlaz) [reply SYCL/XPTI/basic_event_collection.cpp]: I will update the XPTI related tests. To have them smaller.

**Review by alexbatashev — APPROVED** [review]

**Review by bader — APPROVED** [review]

**Pull request merged**